iOS/Extensions/UIImage.swift

```diff
@@ -16,8 +16,23 @@ extension UIImage {
         return CGSize(width: pageSize.width, height: scaledHeight)
     }
 
-    func saveToAlbum(_ name: String? = nil) {
-        let albumName = name ?? Bundle.main.object(forInfoDictionaryKey: "CFBundleDisplayName") as? String ?? "Aidoku"
+    func saveToAlbum(_ name: String? = nil, viewController: BaseViewController) {
+        let status = PHPhotoLibrary.authorizationStatus(for: .readWrite)
+        if status == .restricted || status == .denied {
+            viewController.confirmAction(
+                title: NSLocalizedString("ENABLE_PERMISSION", comment: ""),
+                message: NSLocalizedString("PHOTOS_ACCESS_DENIED_TEXT", comment: ""),
+                continueActionName: NSLocalizedString("SETTINGS", comment: "")
+            ) {
+                if let settings = URL(string: UIApplication.openSettingsURLString) {
+                    UIApplication.shared.open(settings)
+                }
+            }
+        }
```

Comment:
should this just stop here if there’s no permission? the code after this would fail anyways, right?
```suggestion
        guard status != .restricted && status != .denied else {
            viewController.confirmAction(
                title: NSLocalizedString("ENABLE_PERMISSION", comment: ""),
                message: NSLocalizedString("PHOTOS_ACCESS_DENIED_TEXT", comment: ""),
                continueActionName: NSLocalizedString("SETTINGS", comment: "")
            ) {
                if let settings = URL(string: UIApplication.openSettingsURLString) {
                    UIApplication.shared.open(settings)
                }
            }
            return
        }
```

Reply:
yes